### ISSUE TITLE
qt mp4 support

### ISSUE THREAD

**Issue by mllapps** [pbouda/buildroot-qt-dev#4]
Have you tested the multimedia? I tried the media player sample but it doesn't work.

**Comment by pbouda** [comment]
For me multimedia works, though I did not try the sample... I use it in this project: http://brummbeere.readthedocs.org/en/latest/

Do you have any logs?

**Comment by mllapps** [comment]
I am not sure but it say that no video sink is available. I will try a fresh build for raspberry pi B+. Then I copy the console output.

**Comment by mllapps** [comment]
Does brummbeere support video decoding? If I understand right then the brummbeere project is a audio player?

**Comment by pbouda** [comment]
Yes, you are a right, it's audio only. I did not use video yet, so in fact there might be a video sink missing.

**Comment by mllapps** [comment]
Do you have an idea what I have to enable to get video support?

**Comment by pbouda** [comment]
It should be in the gstreamer-plugins packages. You should check which video sinks there are and select the ones you need.

**Comment by mllapps** [comment]
All plugin options are selected but I think the `BR2_PACKAGE_GST1_PLUGINS_BAD_PLUGIN_FBDEV` is the right one. The help information is  `Linux framebuffer video sink`. But I don't know how to select the video sink at Qt source code.

**Comment by mllapps** [comment]
Currently I Debug the buildroot steps for gstreamer-1.0 and the fbdevsink should be available. But I think some for qt goes wrong.

```
configure: *** checking feature: linux framebuffer ***
configure: *** for plug-ins: fbdevsink ***
checking linux/fb.h usability... yes
checking linux/fb.h presence... yes
checking for linux/fb.h... yes
configure: *** These plugins will be built: fbdevsink
...
...
configure: *** checking feature: Qt elements ***
configure: *** for plug-ins: qt ***
configure: *** These plugins will not be built: qt
...
...
```

**Comment by pbouda** [comment]
I think framebuffer won't work, as the Qt here uses eglfs for rendering. I found this, might be relevant:

http://gstreamer-devel.966125.n4.nabble.com/Building-Qt-plugin-for-embedded-i-MX6-td4674283.html

**Comment by mllapps** [comment]
I am confused because there are so many forum threads on the net about the Qt5 and gstreamer problem on embedded Linux devices. A really interesting part is https://gkiagia.wordpress.com/tag/qt-gstreamer/

https://github.com/GStreamer/qt-gstreamer

**Comment by mllapps** [comment]
OK currently I tested the mp4 support with the original Linux (raspbian) for the raspberry pi. The tutorial I used is https://wiki.qt.io/RaspberryPi2EGLFS. 

**Conclusion**
Now it works with Qt5.6.1 but not very stable. Sometimes the audio starts but the video has a still image. But I learned something so in the feature I will try to configure it for buildroot.

The OpenGL tests are better. I got 60fps with 9% CPU usage. If the touchscreen works the fps goes down to 30-40fps while I am moving the fingers. 

I think I know the missing settings to support OpenGL with the QMultimedia framework. If I had time then I will try to patch this.

@pbouda You are right with the eglfs support. LinuxFB is not supported for hardware rendering. http://doc.qt.io/qt-5/embedded-linux.html#configuring-a-specific-device

**Comment by mllapps** [comment]
Now OpenGL works on the raspberry pi 2 with Qt5 on my custom buildroot setup.  

Roadmap: 
- integrate omxplayer
- finish my omxplayer frontend (Qt5)
- try gstreamer with Qt5 and video support
